src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Colors = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Rectangle = 'rectangle',
+  Circle = 'circle',
 }
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape;
+  color: Colors;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+
+    sides.sort((x, y) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('Attention, your triangle is not correct');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Attention, your triangle is not correct');
+    }
+  }
 
+  getArea(): number {
+    const square = (this.a + this.b + this.c) / 2;
```

Comment:
make sense to destructure a, b, and c here
for example: 
`const { a, b, c } = this;`